src/Centreon/Domain/PlatformTopology/PlatformTopologyService.php

```diff
@@ -468,11 +477,11 @@ private function setMonitoringServerId(Platform $platform): void
     /**
      * Search for already registered platforms using same name or address
      *
-     * @param Platform $platform
+     * @param PlatformInterface $platform
      * @throws PlatformTopologyConflictException
      * @throws EntityNotFoundException
      */
-    private function checkForAlreadyRegisteredSameNameOrAddress(Platform $platform): void
+    private function checkForAlreadyRegisteredSameNameOrAddress(PlatformInterface $platform): void
```

Comment:
As mention before, you could use this method to do a findPlatformByAddress & findPlatformByName instead of calling isPlatformAlreadyRegisteredInTopology l.499

Reply:
$this->platformTopologyRepository->isPlatformAlreadyRegisteredInTopology to split

Reply:
done